tests/mu_define_enum_test.c

```diff
@@ -34,5 +75,76 @@ int run_mu_define_enum_tests(void)
+
+    const char* TEST6_test6_a = MU_ENUM_TO_STRING(TEST6_ENUM, test6_a);
+    POOR_MANS_ASSERT(TEST6_test6_a != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_a = 42", TEST6_test6_a) == 0);
+
+    // test6_b and TEST6_ENUM_INVALID are both 0
+    const char* TEST6_test6_b = MU_ENUM_TO_STRING(TEST6_ENUM, test6_b);
+    POOR_MANS_ASSERT(TEST6_test6_b != NULL);
+    POOR_MANS_ASSERT(strcmp("TEST6_ENUM_INVALID", TEST6_test6_b) == 0);
+
+    const char* TEST6_test6_c = MU_ENUM_TO_STRING(TEST6_ENUM, test6_c);
+    POOR_MANS_ASSERT(TEST6_test6_c != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_c = -1", TEST6_test6_c) == 0);
+
+    // test6_d and TEST6_ENUM_INVALID are both 0
+    const char* TEST6_test6_d = MU_ENUM_TO_STRING(TEST6_ENUM, test6_d);
+    POOR_MANS_ASSERT(TEST6_test6_d != NULL);
+    POOR_MANS_ASSERT(strcmp("TEST6_ENUM_INVALID", TEST6_test6_d) == 0);
+
+    const char* TEST6_test6_e = MU_ENUM_TO_STRING(TEST6_ENUM, test6_e);
+    POOR_MANS_ASSERT(TEST6_test6_e != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_e", TEST6_test6_e) == 0);
+
+    // test6_e and test6_f are both the same value
+    const char* TEST6_test6_f = MU_ENUM_TO_STRING(TEST6_ENUM, test6_f);
+    POOR_MANS_ASSERT(TEST6_test6_f != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_e", TEST6_test6_f) == 0);
+
+    const char* TEST6_test6_g = MU_ENUM_TO_STRING(TEST6_ENUM, test6_g);
+    POOR_MANS_ASSERT(TEST6_test6_g != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_g = test6_f + 1", TEST6_test6_g) == 0);
+
+    const char* TEST6_test6_i = MU_ENUM_TO_STRING(TEST6_ENUM, test6_i);
+    POOR_MANS_ASSERT(TEST6_test6_i != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_i = 5 - 1", TEST6_test6_i) == 0);
+
+    const char* TEST6_test6_j = MU_ENUM_TO_STRING(TEST6_ENUM, test6_j);
+    POOR_MANS_ASSERT(TEST6_test6_j != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_j = -2", TEST6_test6_j) == 0);
+
+    POOR_MANS_ASSERT(MU_ENUM_VALUE_COUNT_WITHOUT_INVALID(TEST7_ENUM_VALUES) == 2);
+
+    const char* TEST7_test7_a = MU_ENUM_TO_STRING(TEST7_ENUM, 0);
+    POOR_MANS_ASSERT(TEST7_test7_a != NULL);
+    POOR_MANS_ASSERT(strcmp("test7_a", TEST7_test7_a) == 0);
+
+    const char* TEST7_test7_b = MU_ENUM_TO_STRING(TEST7_ENUM, 42);
+    POOR_MANS_ASSERT(TEST7_test7_b != NULL);
+    POOR_MANS_ASSERT(strcmp("test7_b = 42", TEST7_test7_b) == 0);
+
+    POOR_MANS_ASSERT(MU_ENUM_VALUE_COUNT_WITHOUT_INVALID(TEST8_ENUM_VALUES) == 3);
+
+    const char* TEST8_test8_a = MU_ENUM_TO_STRING(TEST8_ENUM, -1);
+    POOR_MANS_ASSERT(TEST8_test8_a != NULL);
+    POOR_MANS_ASSERT(strcmp("test8_a = -1", TEST8_test8_a) == 0);
+
+    const char* TEST8_test8_b = MU_ENUM_TO_STRING(TEST8_ENUM, 5);
+    POOR_MANS_ASSERT(TEST8_test8_b != NULL);
+    POOR_MANS_ASSERT(strcmp("test8_b = 5", TEST8_test8_b) == 0);
+
+    const char* TEST8_test8_c = MU_ENUM_TO_STRING(TEST8_ENUM, 0);
+    POOR_MANS_ASSERT(TEST8_test8_c != NULL);
+    POOR_MANS_ASSERT(strcmp("test8_c = 0", TEST8_test8_c) == 0);
+
     return result;
```

Comment:
>result [](http://example.com/codeflow?start=11&length=6)

We should have some tests that exercise the caching of "has_equal". That is, stringify twice the same enum - does it still works?

Reply:
Added!

---
In reply to: [632265230](https://github.com/Azure/macro-utils-c/pull/30#discussion_r632265230) [](ancestors = 632265230)